508 imagelib search php

NikitaSalikov opened the pull request:
Added Doctype and primary language at the top
Lines 11-22 changed FILTER to htmlspecialchar Created labels for taxontype( 104), phuid (122), tags(139), imagecount(167), imagetype(182), 

Additionally:

Content/imagelib/search.en.php - added strings in English to labeled values
Content/imagelib/search.es.php - added strings in Spanish to labeled values (Not sure if correct)
Content/imagelib/search.fr.php - added strings in French to labeled values(Not sure if correct)
classes/ImageLibrarySearch - lines 393 , 422, 430 changed FILTER to htmlspecialchar classes/OccurenceTaxaManager - line 379 changed FILTER to htmlspecialchar.

Pull Request Checklist:

- [X] Features and backlog bugs should be merged into the `Development` branch, **NOT** `master`
- [N/A] Hotfixes should be merged into both the `Development` and `master` branches (at the same time).
- [X] All new text is preferrably internationalized (i.e., no end-user-visible text is hard-coded on the PHP pages)
- [X] There are no linter errors
- [N/A] New features have responsive design (i.e., look aesthetically pleasing both full screen and with small or mobile screens)
- [X] [Symbiota coding standards](https://docs.google.com/document/d/1-FwCZP5Zu4f-bPwsKeVVsZErytALOJyA2szjbfSUjmc/edit?usp=sharing) have been followed
- [N/A] If any files have been reformatted (e.g., by an autoformatter), the reformat is its own, separate commit in the PR
- [X] Comment which GitHub issue(s), if any does this PR address
- [X] It is the code author's responsibility to merge their own pull request after it has been approved
- [X] If this PR represents a merge into the `Development` branch, remember to use the **squash & merge** option
- [N/A] If this PR represents a merge from the `Development` branch into the master branch, remember to use the **merge** option
- [X] If the dev team has agreed that this PR represents the last PR going into the Development branch before a tagged release (i.e., before an imminent merge into the master branch), make sure to notify the team and [lock the `Development` branch](https://github.com/BioKIC/Symbiota/settings/branches) to prevent accidental merges while QA takes place

Thanks for contributing and keeping it clean!

NikitaSalikov committed:
updated index.php file according to 508 compliance

NikitaSalikov committed:
fixes for imagelib/search

NikitaSalikov committed:
Merge branch 'Development' into 508-imagelib-search-php

Atticus29 commented on imagelib/search.php:
```diff
@@ -1,3 +1,7 @@
+<!DOCTYPE html>
+
+<html lang="<?php echo $LANG_TAG ?>">
```
There's an <html> tag in ln. ~43. It would be better to add the lang attribute there instead and remove this, I think.

Atticus29 reviewed (changes requested):
All of the current changes look great! Thank you for helping with all of that!

I think there are a few more to do, though:

<img width="1552" alt="Screen Shot 2023-07-17 at 10 40 14 AM" src="https://github.com/BioKIC/Symbiota/assets/2775448/d532331e-2475-4094-87a7-220b37b88a0c">

I think we should maybe add a "Taxon" label to this input:
<img width="1226" alt="Screen Shot 2023-07-17 at 12 58 01 PM" src="https://github.com/BioKIC/Symbiota/assets/2775448/8210e87f-b816-4536-ba52-4f9b13ea2140">


These should be resolved:

<img width="948" alt="Screen Shot 2023-07-17 at 12 59 40 PM" src="https://github.com/BioKIC/Symbiota/assets/2775448/203f70ad-56c4-42a2-b780-86268a5297d6">
<img width="951" alt="Screen Shot 2023-07-17 at 1 00 48 PM" src="https://github.com/BioKIC/Symbiota/assets/2775448/ac925958-cb23-42fa-92a0-c213ac94bee8">

<img width="768" alt="Screen Shot 2023-07-17 at 1 10 03 PM" src="https://github.com/BioKIC/Symbiota/assets/2775448/f12109ab-8bc6-40b9-a6b3-12deabdf3e0d">

Finally, there's the onChange issue here:

<img width="942" alt="Screen Shot 2023-07-17 at 1 15 17 PM" src="https://github.com/BioKIC/Symbiota/assets/2775448/ec662395-419a-4c67-b317-4c2d70431f58">

I'd recommend implementing an onChange and an onKeypress instead of just the onChange. I believe that I implemented this in a different PR. If I forget to follow up and post that in a comment here, please ask me about it!

NikitaSalikov committed:
fixed the remaining TotalValidator errors

NikitaSalikov committed:
508 Compliance fixes

NikitaSalikov committed:
Resolve merge conflicts

NikitaSalikov committed:
Quick fix of taxon type and taxon labels

NikitaSalikov commented:
Hey! 

Ready for review. Here is what I did:

Line 43: Moved Lang attribute from top
Line 112, 116-117: Added Labels to input “taxa”, “usethes” and LANG lines “Search by” and “Include Synonyms” in lang/imagelb/search.* (en/es/fr) in respective languages.
Line 123,140,183: Added id for select

(!)Line179: Added onKeypress. Not sure if implementation is correct.

Atticus29 reviewed (commented):
Nice job, @NikitaSalikov !

I re-ran TotalValidatorPro, and it seems happy with the changes except for for onchange/onkeypress for the `<select>` element.

It also stil wants you to either add text to `<meta name='keywords' content='' />` or to remove the `content=''` attribute.

In terms of the onchange listener, I think that our solution needs to be a little more thorough. I believe that this warrants a separate ticket, so don't worry about fixing that for this current ticket. If you fix the `content=''` issue, I'll approve this PR.

NikitaSalikov committed:
fixed the remaining TotalValidator errors

Atticus29 reviewed (approved):
LGTM! Thanks!

NikitaSalikov committed:
Merge branch 'Development' into 508-imagelib-search-php
